tests/test_packaging.py

```diff
@@ -2,11 +2,14 @@
 
 import os.path
 import pathlib
-from typing import List
+from pprint import pformat as pf
+from typing import Final, List
 
 import pytest
 import requirements as rp
 
+IGNORE_PINS: Final[set[str]] = {"mypy", "ruff", "pytest"}
```

Comment:
It doesn't make sense to check some of these development only packages. These packages SHOULD be pinned, but these checks punish pinning and make it very tedious to update the versions/pins.

I've added this `IGNORE_PINS` constant so that the packaging step just ignores anything here.

I've also had it completely ignore the `requirement-types` package since most of those types are either stubs or packages that are pulled from other requirement files which are already checked here.